src/librbd/internal.cc

```diff
@@ -1943,6 +1943,19 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
 	return r;
       }
 
+      if (!old_format) {
+        r = mirror_image_disable_internal(ictx, true);
+        if (r < 0) {
+          lderr(cct) << "error disabling image mirroring: " << cpp_strerror(r)
+                     << dendl;
+          ictx->owner_lock.put_read();
+          ictx->state->close();
+          return r;
+        }
+
+        // TODO: http://tracker.ceph.com/issues/14421
```

Comment:
Correct -- we cannot assume that the local librbd has access to the remote peer clusters. We will let the remote rbd-mirror daemon "pull" the deletion.